packages/react/src/Grid/Grid.tsx

```diff
@@ -7,16 +7,11 @@ import clsx from 'clsx'
 import { forwardRef } from 'react'
 import type { ForwardedRef, HTMLAttributes, PropsWithChildren } from 'react'
 import { GridCell } from './GridCell'
+import { paddingClasses } from './paddingClasses'
 
 export type GridColumnNumber = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12
-
-export type GridColumnNumbers = {
-  narrow: GridColumnNumber
-  medium: GridColumnNumber
-  wide: GridColumnNumber
-}
-
 export type GridPaddingSize = 'small' | 'medium' | 'large'
+export type ResponsiveGridValues<T> = T | { narrow: T; medium: T; wide: T }
```

Comment:
This creates a type with the same column count for all sizes, no? I don't think that's what we want.